cpp/unit/ability.cpp

```diff
@@ -21,6 +21,16 @@ bool has_resource(Unit &target) {
 	       target.get_attribute<attr_type::resource>().amount > 0;
 }
 
+bool is_ally(Unit &to_modify, Unit &target) {
+	if (to_modify.has_attribute(attr_type::owner) &&
+		target.has_attribute(attr_type::owner)) {
+		auto &mod_player = to_modify.get_attribute<attr_type::owner>().player;
+		auto &tar_player = target.get_attribute<attr_type::owner>().player;
+		return mod_player.is_ally(tar_player);
+	}
+	return false;
```

Comment:
sure it's a good idea to say "it's not an ally" when one of the units doesn't have the owner attribute?